tests/main/hidden-snap-dir/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: Check that the experimental hidden dir feature migrates the dir
+
+environment:
+    NAME: test-snapd-tools
+
+prepare: |
+    snap pack "$TESTSLIB"/snaps/"$NAME"
+    "$TESTSTOOLS"/snaps-state install-local "$NAME"
+
+restore: |
+    snap set system experimental.hidden-snap-folder!
```

Comment:
(nitpick) `snap unset  system experimental.hidden-snap-folder` is maybe a bit easier to read